numpy/__init__.pyi

```diff
@@ -3927,7 +3925,9 @@ class poly1d:
         k: None | _ArrayLikeComplex_co | _ArrayLikeObject_co = ...,
     ) -> poly1d: ...
 
-class matrix(ndarray[_ShapeType, _DType_co]):
+
+_ShapeType_2D_co = TypeVar("_ShapeType_2D_co", covariant=True, bound=tuple[int, int])
```

Comment:
I think it makes sense to move this towards the other `_ShapeType` type vars.

But you can leave it here as well, although you should then add a newline after it, since classes in `.pyi` files require at least 1 newline before/after it (if I remember correctly).

Reply:
The current name might also be a bit confusing with `_ShapeType2`.
Renaming it to something like `_MatrixShapeType_co` or `_Shape2DType_co` could help with this.